docs/specs/lookup.md

```diff
@@ -0,0 +1,31 @@
+# Product Requirements Specification
+
+Implement fast lookup via primary key or secondary indices
+
+## Problem:
+
+There are use cases within big-data scenarios where fast lookup with index are beneficial. 
+Especially in streaming processing, lookup latency have a significant impact to overall
+throughput. TiBigData implements full table scan only at this time, which makes it suboptimal to
+workloads that requires low latency. Therefore, flink-tidb-connector utilize JDBC lookup table 
+source instead. Since Flink, like other big-data computation frameworks, is at computing layer by
+itself. Sending lookup request to TiDB server which is at computing layer as well is redundant and 
+waste of resources. Lookup with TiDB server contributes to growing of latency with extra hops in
+access path.
+
+## Goals:
+
+Implement primary key and secondary index based lookup in TiBigData. Decreases lookup latency with
+the best efforts and saves unnecessary resources waste for load balancers and TiDB servers.
```

Comment:
the unnecessary waste of resources

Reply:
Changed from `waste of resources` to `costs`.